### ISSUE TITLE
Nunjucks example not passing watch configuration

### ISSUE THREAD

**Issue by cyberwombat** [hapijs/vision#75]
Trying to setup Nunjucks - set `watch` to true in the prepare options as outlined in your example but it will not watch/recompile. Has this example been tested?

**Comment by cyberwombat** [comment]
Ok. So watch is only called when `env.render` is called - not `nunjucks.render`. Also `env.render` seems to want to the filename directly so the working code is:
````
 var env = null;
  server.views({
    engines: {
      html: {
        compile: function(src, options) {
          return function(context) {            
            return env.render(options.filename. context);
          };
        },

        prepare: function(options, next) {

         env = Nunjucks.configure(options.path, {
            watch: true,
            noCache: true
          });
          return next();
        }
      }
    },
    path: Path.join(__dirname, 'views')
  });
`````
However its a bit unclean with env variable. Is there a way to pass the env created in `prepare` to `compile`? and since nujucks must read the files then its not necessary for vision to read it and pass the `src` - how can this be optimized?

**Comment by jagoda** [comment]
The example does work as specified. In fact, there is an executable version in the `examples` directory that you can run for yourself. However, you will note that the example has `watch` set to `false`. This is intentional because `vision` caches compiled templates by default (making `watch` somewhat pointless). If you reexamine the example I think you will find that an `Environment` instance is already being added to the compile context via the `compileOptions` attribute and is being leveraged during template compilation. `vision` does not currently allow for passing the filename to a template engine without loading the template source from disk. This is because this is how most template engines work. Nunjucks supports this use case just fine using the `compile()` method rather than the `render()` method.

In short, I believe that you can get the behavior that you want by simply disabling template caching in `vision` (see the [docs](https://github.com/hapijs/vision/blob/master/API.md#serverviewsoptions)) and using the unmodified example code for integrating with nunjucks.

**Comment by jagoda** [comment]
Does this address your questions?

**Comment by cyberwombat** [comment]
Yes thank you.

**Comment by FriseHunter** [comment]
Good afternoon, I have a similar problem, I don’t know how to set up a watch for nunjucks under a webpack, webpack listens to everything (css, js), but not .njk

I use nunjucks-webpack-plugin and nunjucks-webpack-loader in my build.

Please help me figure it out.
`const path = require('path');
const ExtractTextPlugin = require("extract-text-webpack-plugin");
const CopyPlugin = require('copy-webpack-plugin');
const CleanWebpackPlagin = require('clean-webpack-plugin');
const autoprefixer = require('autoprefixer');
const RemoveStrictPlugin = require('remove-strict-webpack-plugin');
const NunjucksWebpackPlugin = require("nunjucks-webpack-plugin");
const SvgStore = require('webpack-svgstore-plugin');

let config = {
    entry: './src/index.js',
    output: {
        path: path.resolve(__dirname, 'build'),
        filename: 'js/main.js',
        publicPath: '/'
    },
    devServer: {
        overlay: {
            warnings: true,
            errors: true
        }
    },
    module: {
        rules: [{
                test: /\.(njk|nunjucks)$/,
                use: {
                    loader: 'nunjucks-webpack-loader',
                    options: {
                        watch: true,
                    },
                },
                test: /\.m?js$/,
                exclude: /(node_modules)/,
                use: {
                    loader: 'babel-loader',
                    options: {
                        presets: ['@babel/preset-env']
                    }
                },
                test: /\.scss$/,
                use: ExtractTextPlugin.extract({
                    use: [{
                            loader: "css-loader",
                            options: {
                                sourceMap: true
                            }
                        },
                        {
                            loader: 'postcss-loader',
                            options: {
                                plugins: [
                                    autoprefixer({
                                        browsers: ['ie >= 8', 'last 4 version']
                                    })
                                ],
                                sourceMap: true
                            }
                        },
                        { loader: "sass-loader" }
                    ]
                }),

            },
            {
                test: /\.(png|jpe?g|gif|ico|svg)(\?v=[a-z0-9]\.[a-z0-9]\.[a-z0-9])?$/,
                use: [{
                        loader: 'file-loader',
                        options: {
                            name: '[path][name].[ext]',
                            outPath: './',
                            useRelativePath: true
                        }
                    },
                    {
                        loader: 'image-webpack-loader',
                        options: {
                            mozjpeg: {
                                progressive: true,
                                quality: 70,
                            }
                        }
                    }
                ]
            },
            {
                test: /\.(woff2|woff|ttf|eot)(\?v=[a-z0-9]\.[a-z0-9]\.[a-z0-9])?$/,
                use: [{
                    loader: 'file-loader',
                    options: {
                        name: 'fonts/[name].[ext]',
                        outPath: './fonts/',
                        useRelativePath: true
                    }
                }]
            },
        ],
    },
    plugins: [
        new RemoveStrictPlugin(),
        new NunjucksWebpackPlugin({
            templates: [{
                from: "./src/index.njk",
                to: "index.html"
            }],
        }),
        new CopyPlugin([{
            from: `./src/img`,
            to: `./img`,
            ignore: ['*.svg']
        }]),
        new SvgStore({
            svgoOptions: {
                plugins: [
                    { removeTitle: true }
                ]
            },
            prefix: ''
        }),
        new ExtractTextPlugin("./styles.css"),
        require('webpack-css-min'),
        new CleanWebpackPlagin(),
    ],
    devtool: 'eval-sourcemap'
};

module.exports = (env, options) => {
    const production = options.mode === 'production';
    config.devtool = production ?
        false : 'eval-sourcemap';

    console.log(options.mode);

    if (options.mode == 'production') {
        config.output.publicPath = './'
    }

    return config
};`

вот мой webpack.config
